nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/operations/document/AddFacet.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * (C) Copyright 2016 Nuxeo SA (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Thibaud Arguillere
+ */
+
+package org.nuxeo.ecm.automation.core.operations.document;
+
+import org.nuxeo.ecm.automation.OperationException;
+import org.nuxeo.ecm.automation.core.Constants;
+import org.nuxeo.ecm.automation.core.annotations.Context;
+import org.nuxeo.ecm.automation.core.annotations.Operation;
+import org.nuxeo.ecm.automation.core.annotations.OperationMethod;
+import org.nuxeo.ecm.automation.core.annotations.Param;
+import org.nuxeo.ecm.automation.core.collectors.DocumentModelCollector;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+
+/**
+ * Operation that adds a facet to the document.
+ *
+ * IMPORTANT: This requires the document to be saved immediately.
+ * @since 8.3
+ */
+@Operation(id = AddFacet.ID, category = Constants.CAT_DOCUMENT, label = "Add Facet", description = "Adds the facet to the document. Notice: The operation saves the document.", aliases = { "Document.AddFacet" })
+public class AddFacet {
+
+    public static final String ID = "Document.AddFacet";
+
+    @Param(name = "facet", required = true)
+    protected String facet = "";
+
+    @Context
+    protected CoreSession session;
+
+    @OperationMethod(collector = DocumentModelCollector.class)
+    public DocumentModel run(DocumentModel input) {
+
+        // we ignore the result
+        /*boolean result = */ input.addFacet(facet);
+
```

Comment:
For such small method (next one in next file too) having extra blank lines doesn't bring anything.